server/paymentProviders/stripe/alipay.ts

```diff
@@ -0,0 +1,214 @@
+      const hostStripeAccount = await order.collective.getHostStripeAccount();
+      const hostPlan = await host.getPlan();
+      const hostFeeSharePercent = hostPlan?.hostFeeSharePercent;
+      const isSharedRevenue = !!hostFeeSharePercent;
+
+      // Read or compute Platform Fee
+      const platformFee = await getPlatformFee(order.totalAmount, order, host, { hostPlan, hostFeeSharePercent });
+      const platformTip = order.data?.platformFee;
+      const intent = await stripe.paymentIntents.retrieve(payment_intent, {
+        stripeAccount: hostStripeAccount.username,
+      });
+
+      const charge = intent.charges.data[0];
+
+      const balanceTransaction = await stripe.balanceTransactions.retrieve(charge.balance_transaction, {
+        stripeAccount: hostStripeAccount.username,
+      });
+
+      // Create a Transaction
+      const fees = extractFees(balanceTransaction, balanceTransaction.currency);
+      const amountInHostCurrency = convertFromStripeAmount(balanceTransaction.currency, balanceTransaction.amount);
+      const hostFeeInHostCurrency = await getHostFee(amountInHostCurrency, order);
+      const data = {
+        charge,
+        balanceTransaction,
+        isFeesOnTop: order.data?.isFeesOnTop,
+        isSharedRevenue,
+        platformFee: platformFee,
+        platformTip,
+        hostFeeSharePercent,
+      };
+      const hostCurrencyFxRate = amountInHostCurrency / order.totalAmount;
+      const platformFeeInHostCurrency = isSharedRevenue ? platformTip * hostCurrencyFxRate || 0 : fees.applicationFee;
+
+      const transactionPayload = {
+        CreatedByUserId: order.CreatedByUserId,
+        FromCollectiveId: order.FromCollectiveId,
+        CollectiveId: order.CollectiveId,
+        PaymentMethodId: order.PaymentMethodId,
+        type: TransactionTypes.CREDIT,
+        OrderId: order.id,
+        amount: order.totalAmount,
+        currency: order.currency,
+        hostCurrency: balanceTransaction.currency.toUpperCase(),
+        amountInHostCurrency,
+        hostCurrencyFxRate,
+        paymentProcessorFeeInHostCurrency: fees.stripeFee,
+        taxAmount: order.taxAmount,
+        description: order.description,
+        hostFeeInHostCurrency,
+        platformFeeInHostCurrency,
+        data,
+      };
+
+      await models.Transaction.createFromContributionPayload(transactionPayload, {
+        isPlatformTipDirectlyCollected: true,
+      });
+      await order.update({ status: 'PAID' });
+
+      res.redirect(`${config.host.website}/${order.collective.slug}/donate/success?OrderId=${OrderId}`);
+    } else if (redirect_status === 'failed') {
+      const id = idDecode(OrderId, IDENTIFIER_TYPES.ORDER);
+      debug(`payment for order ${id} failed, deleting order`);
+      const order = await models.Order.findByPk(id, {
+        include: [{ model: models.Collective, as: 'collective' }],
+      });
+      if (order) {
+        await order.destroy();
```

Comment:
Not sure whether it should be destroyed or rather marked as `ERROR`? In both cases, I think it would be great to have some details about the error in `order.data`.